### DISCUSSION TITLE
Include search of comments

### DISCUSSION THREAD

**Discussion by BobWilkes** [VFPX/GoFish#97] [Ideas]
Based on observed results, comment lines are not included in a search.
I'm in the habit of entering  "*X" at the beginning of a line I want to temporarily 'unhook' while I'm making a  mod and testing. If it's a large block of code, I'll surround it with "*X" and an "IF .f." ... "ENDIF" lines.
Once I'm done, I can easily find the code I no longer need by searching within an object for "*X".
That's OK if there's only one form or object involved, but more often than not, changes are spread throughout many objects.
If GoFish had an 'Include comments' or 'Search only comments' option, my life would be much easier.
Would also be of value if I'm looking for a particular phrase in a comment line

**Comment by Jimrnelson** [comment]
There’s a checkbox on the form for whether you want to search comments or notSent from my iPhoneOn May 26, 2023, at 10:40 AM, BobWilkes ***@***.***> wrote:﻿
Based on observed results, comment lines are not included in a search.
I'm in the habit of entering  "*X" at the beginning of a line I want to temporarily 'unhook' while I'm making a  mod and testing. If it's a large block of code, I'll surround it with "*X" and an "IF .f." ... "ENDIF" lines.
Once I'm done, I can easily find the code I no longer need by searching within an object for "*X".
That's OK if there's only one form or object involved, but more often than not, changes are spread throughout many objects.
If GoFish had an 'Include comments' or 'Search only comments' option, my life would be much easier.
Would also be of value if I'm looking for a particular phrase in a comment line

**Comment by BobWilkes** [comment]
AHA! For some reason, I didn't see that.
TNX.


------ Original Message ------
From "Jimrnelson" ***@***.***>
Date 5/26/2023 10:56:06 AM

>
>There’s a checkbox on the form for whether you want to search comments 
>or not...
>Message ID: 
>***@***.***>
>

-- 
This email has been checked for viruses by AVG antivirus software.
www.avg.com

**Comment by lscheffler** [comment]
@BobWilkes Additional to the *Comments* CheckBox on the form, the *Advanced* form offers the same setting.

**Comment by lscheffler** [comment]
@BobWilkes Hi Bob, maybe a bit late. But if you like to search in comments only, you can use Regular Expressions. I give you the brute force search pattern.
- Go *Advanced* Form, or set in the GoFish form directly.
- Check *Search in comments*
- Check *Regular Expressions*
- For *Search* enter something like `^[ \t]*\*.*Search`, where *Search* is your search string. This says (implicit: per line)
  - from start of line; `^`
  - after any or none space or tab; `[ \t]*`
  - and after an asterix; `\*`
  - and any or none other sign that is not line break or line feed; `.*`
  - find `Search`

It will highlight anything from start to your search string, but for now we have no way to exclude this. One can do, this would need extra coding for *Search in comments only*.

Since we can not search over continued lines, it will not work for line-continued-comments like:
```
*SELECT;
 Field1 from Table 1
```
If you need to look up inline comments too, try something like
```((^[ \t]*\*.*Search)|(\&\&.*Search))```
This means something like:
  - The stuff from above (this is, the stuff before `|`, the *or*)
  - or; from &&; `\&\&` 
  - and any or none other sign that is not line break or line feed; `.*`
  - find `Search`

Logically it will not find comments starting with *NOTE*, but I guess you see how to set up the search pattern.

If you managed to get the string, you might use the **Filter *Within same Procedure*** to filter for additional stuff. Only this will not use Regular expression (yet).

**Reply by Jimrnelson** [reply]
I think there's a logical extension, which removes the need for regular expressions.  Replace the "Comments" checkbox so that there are three choices:

- All code and comments
- All code, not comments
- Comments only

**Comment by lscheffler** [comment]
@Jimrnelson Something the like. I was just answering the question. I move this to a distinct issue. See #150
